plugins/check_http.c

```diff
@@ -152,6 +152,8 @@ void print_help (void);
 void print_usage (void);
 char *unchunk_content(const char *content);
 
+extern int check_hostname;
```

Comment:
Sorry, but I think using external variables is making the code overly complex and unreadable.

Also this is a boolean and therefore should be `bool` ( or `_Bool`)